### ISSUE TITLE
Error running tests after Java upgrade to IBM Semeru jdk-17.0.6+10 (from jdk1.8.0_311.jdk)

### ISSUE THREAD

**Issue by thecrookster** [easymock/easymock#393]
With easymock 5.1.0, I had 7 errors, after trying out 5.2.0-SNAPSHOT it dropped to 4.
```
[ERROR] Errors: 
[ERROR]   CARootFolderTest.getExtensionsFolderPath:89->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.glass.services.UserProfileSettings$$$EasyMock$3.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   CARootFolderTest.getExtensionsFolderPathUPSException:101->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.glass.services.UserProfileSettings$$$EasyMock$3.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   CMStoreServiceTest.getCMStoreParametersFieldMockSuccess:94->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.json.JsonObject$$$EasyMock$2.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   CMStoreServiceTest.getCMStoreParametersFieldSuccess:117->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.json.JsonObject$$$EasyMock$2.$callback/org.easymock.internal.ClassMockingData/putField
[INFO] 
[ERROR] Tests run: 281, Failures: 1, Errors: 4, Skipped: 0
```

This is with 5.1.0 (with 7 errors)
```
[ERROR] Errors: 
[ERROR]   CARootFolderTest.getExtensionsFolderPath:89->EasyMockSupport.mock:74 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.glass.services.UserProfileSettings$$$EasyMock$3.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   CARootFolderTest.getExtensionsFolderPathUPSException:101->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.glass.services.UserProfileSettings$$$EasyMock$3.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   SSOContextTest.testGetSSOContext_InitialContext:148 » Runtime java.lang.IllegalAccessException: no such field: javax.naming.InitialContext$$$EasyMock$44.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   SSOContextTest.testGetSSOContext_InitialContext_and_config:170 » Runtime java.lang.IllegalAccessException: no such field: javax.naming.InitialContext$$$EasyMock$44.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   SSOContextTest.testGetSSOContext_InitialContext_none:128 » Runtime java.lang.IllegalAccessException: no such field: javax.naming.InitialContext$$$EasyMock$44.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   CMStoreServiceTest.getCMStoreParametersFieldMockSuccess:94->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.json.JsonObject$$$EasyMock$2.$callback/org.easymock.internal.ClassMockingData/putField
[ERROR]   CMStoreServiceTest.getCMStoreParametersFieldSuccess:117->EasyMockSupport.createStrictMock:321->EasyMockSupport.strictMock:169 » Runtime java.lang.IllegalAccessException: no such field: com.ibm.bi.json.JsonObject$$$EasyMock$2.$callback/org.easymock.internal.ClassMockingData/putField
[INFO] 
[ERROR] Tests run: 281, Failures: 1, Errors: 7, Skipped: 0
```

Appreciate any help on the remaining 4.

This might be related: https://github.com/easymock/easymock/issues/274

Thx
Shawn Crook
IBM

**Comment by henri-tremblay** [comment]
Thanks for reporting. Is it working on a Temurin JVM? And can you provide a simple failing testcase? That would be helpful to quickly reproduce.

**Comment by thecrookster** [comment]
Sample test snippit

```
import javax.naming.InitialContext;

import org.easymock.EasyMock;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class SSOContextTest {
	
	@Before
	public void setup() {
	}
	
	@After
	public void teardown() {
	}
	
	@Test
	public void myFailingTest() throws Exception {
		InitialContext context = EasyMock.createMock(InitialContext.class); //Problem
	}
```

**Comment by thecrookster** [comment]
What PR in 5.2.0 fixes my issue? Thx
Can I overcome this issue without upgrading to 5.2.0? If so... please let m know how. thx.

**Comment by thecrookster** [comment]
@henri-tremblay When will 5.2.0 ship? Thx

**Comment by henri-tremblay** [comment]
Hi. As soon I can.

**Comment by basil** [comment]
Friendly ping that a release of 5.2.0 with Java 21 support would be much appreciated. Thank you for your great work maintaining this component!

**Comment by henri-tremblay** [comment]
Most probably fixed by #442

**Comment by henri-tremblay** [comment]
It's out. Plese give it a spin.
